samza-sql-shell/src/main/java/org/apache/samza/sql/client/interfaces/CommandType.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.sql.client.interfaces;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public interface CommandType {
+
+  static List<String> getAllCommands() { return new ArrayList<>(); }
+
+  String getCommandName();
+
+  String getDescription();
+
+  String getUsage();
+
+  boolean isNoArgsCommand();
```

Comment:
The name of `isNorArgsCommand` could be confusing when a command that takes either parameters or no parameters is added in the future, like the `HELP`. By intuition you never know if that command shall be counted as a no args command or not. A dev may have to check how this flag is used in the code to decide which of `true` and `false` is appropriate.  Maybe a name like `mustHaveArgs`, `isNoArgsOK`, or 'canHaveNoArgs' would be clearer?

Reply:
renamed to argsAreOptional (both member var and fun)